modules/home/transformations/generate-sidebar-upgrade.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace elementor\modules\home\transformations;
+
+use Elementor\Modules\Home\Transformations\Base\Transformations_Abstract;
+use Elementor\Utils;
+
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Generate_Sidebar_Upgrade extends Transformations_Abstract {
+	public bool $has_pro;
+
+	public function __construct( $args ) {
+		parent::__construct( $args );
+
+		$this->has_pro = Utils::has_pro();
+	}
+
+	public function transform( array $home_screen_data ): array {
+		foreach ( $home_screen_data['sidebar_upgrade'] as $index => $item ) {
+			if ( $item['header']['title'] === "" ) {
+				unset( $home_screen_data['sidebar_upgrade'] );
+				break;
+			}
+			if ( $this->has_pro && $item['license'][0] === 'free' ) {
+				unset( $home_screen_data['sidebar_upgrade'][$index] );
+				break;
+			} elseif ( ! $this->has_pro && count( $item['license'] ) > 1 ) {
```

Comment:
We would like to test for the specific license, so that product can create different content for e.g. the free, essential, advanced and hosting plans, if they would like to do so.

Reply:
See https://github.com/elementor/elementor/blob/7495c9f096cbf9f4513f863fad81d3a73077c28e/core/common/modules/connect/module.php#L24 for reference.

Reply:
I think splitting it license by license could be a separate task, also for now we wanted to control pro & core separately.